ext/mysqlnd/mysqlnd_enum_n_def.h

```diff
@@ -348,14 +348,27 @@ typedef enum mysqlnd_server_option
 #define SET_FLAG				 2048
 #define NO_DEFAULT_VALUE_FLAG	 4096
 #define ON_UPDATE_NOW_FLAG		 8192
+#define NUM_FLAG				32768
+
+/* The following flags are marked as internal in mysql_com.h */
 #define PART_KEY_FLAG			16384
 #define GROUP_FLAG				32768
-#define NUM_FLAG				32768
 
-#define IS_PRI_KEY(n)	((n) & PRI_KEY_FLAG)
-#define IS_NOT_NULL(n)	((n) & NOT_NULL_FLAG)
-#define IS_BLOB(n)		((n) & BLOB_FLAG)
-#define IS_NUM(t)		((t) <= FIELD_TYPE_INT24 || (t) == FIELD_TYPE_YEAR || (t) == FIELD_TYPE_NEWDECIMAL)
```

Comment:
`IS_NUM` condition has changed. What impact does the removed condition have?
In other words, what happens in the new logic if the value is `FIELD_TYPE_YEAR` or `FIELD_TYPE_NEWDECIMAL`?

Reply:
IS_NUM was declared, but never used by mysqlnd. I suppose this would be a breaking change though for anything that imports this header file externally and uses this definition. Internally, mysqlnd sets the NUM_FLAG under the following conditions:
https://github.com/php/php-src/blob/0e33b8dad9c376b14bbaf2ac3049e50833d2e5ca/ext/mysqlnd/mysqlnd_wireprotocol.c#L1221-L1228

However, MySql sets the NUM_FLAG based on their definition of IS_NUM here:
https://github.com/mysql/mysql-server/blob/596f0d238489a9cf9f43ce1ff905984f58d227b6/include/mysql.h#L116-L118

Maybe the condition in mysqlnd should be changed to use the condition provided by mysql?